### PR TITLE
xbase: add retry times for set listener #124

### PR DESCRIPTION
**Pull request by zhyass**
[summary]
We want to use the publishNotReadyAddresses parameter in kubernetes to be able to access pods even if the pod is not ready.
But even if this parameter is used, we may still need to wait 30s after pod creation. So we add retrys 30s for set the listener.

Just merge into the branch feature_adapt_k8s

### CODE REVIEW

**Comment by drdstech** [comment]
LGTM

**Review by andyli029 — COMMENTED** [review]
publishNotReadyAddresses
https://github.com/andyli029/krypton-helm/pull/35

File: src/xbase/xrpc/xrpc.go [hunk 1]
```diff
@@ -49,26 +49,39 @@ func (s *Service) Start() error {
...
 			}
 			go s.server.ServeConn(conn)
 		}
 	}()
-	s.opts.Log.Warning("xrpc.Start.listening.on[%v]", lis.Addr())
+	s.opts.Log.Warning("xrpc.Start.listening.on[%v]", s.listener.Addr())
 	return nil
 }
 
+func (s *Service) setListener() error {
```
Comment (hustjieke) [inline src/xbase/xrpc/xrpc.go:71]: Maybe we use a Time.Ticker here will be better ? e.g.:
 go func() {
       for range ticker {
               event1
               event2
               ...
}
}
select {
case event1:
...
case event2:
...
}

Comment (zhyass) [reply src/xbase/xrpc/xrpc.go:71]: This is not a scheduled task. The logic here is to set the listener. If announce listener fail, wait for 1s, and then try again. A total of 30 times. If more than 30 times, return error.
Only if announce listener success, the following steps can be carried out.

**Comment by hustjieke** [comment]
LGTM

**Comment by andyli029** [comment]
LGTM

**Pull request merged**